File: internal/modules/api/dto/auth_dto.go
package dto

import "time"

// RegisterRequest represents user registration request.
type RegisterRequest struct {
	Email    string `json:"email" validate:"required,email"`
	Password string `json:"password" validate:"required,min=8"`
	Name     string `json:"name" validate:"required,min=2"`
}

// LoginRequest represents user login request.
type LoginRequest struct {
	Email    string `json:"email" validate:"required,email"`
	Password string `json:"password" validate:"required"`
}

// RefreshRequest represents token refresh request.
type RefreshRequest struct {
	RefreshToken string `json:"refresh_token" validate:"required"`
}

// AuthResponse represents authentication response with tokens.
type AuthResponse struct {
	AccessToken  string       `json:"access_token"`
	RefreshToken string       `json:"refresh_token"`
	ExpiresIn    int64        `json:"expires_in"` // seconds
	User         UserResponse `json:"user"`
}

// UserResponse represents user data in responses.
type UserResponse struct {
	ID               string     `json:"id"`
	Email            string     `json:"email"`
	Name             string     `json:"name"`
	PlayerID         *string    `json:"player_id,omitempty"`
	SubscriptionTier string     `json:"subscription_tier"`
	SubscriptionExpiresAt *time.Time `json:"subscription_expires_at,omitempty"`
	EmailVerified    bool       `json:"email_verified"`
	CreatedAt        time.Time  `json:"created_at"`
}

// LinkPlayerRequest represents request to link user to player.
type LinkPlayerRequest struct {
	PlayerID string `json:"player_id" validate:"required"`
}

// UpdateProfileRequest represents profile update request.
type UpdateProfileRequest struct {
	Name string `json:"name" validate:"omitempty,min=2"`
}

File: ios/StarRink/StarRink/Data/DTOs/AuthDTO.swift
import Foundation

struct LoginRequest: Encodable {
    let email: String
    let password: String
}

struct RegisterRequest: Encodable {
    let email: String
    let password: String
    let name: String
}

struct RefreshRequest: Encodable {
    let refreshToken: String
}

struct LinkPlayerRequest: Encodable {
    let playerId: String
}

struct AuthResponse: Decodable {
    let accessToken: String
    let refreshToken: String
    let expiresIn: Int
    let user: UserDTO
}

struct UserDTO: Decodable {
    let id: String
    let email: String
    let name: String
    let role: String?
    let subscriptionTier: String
    let playerId: String?
    let emailVerified: Bool
    let createdAt: Date?
}

typealias UserResponse = UserDTO
